### ISSUE TITLE
Update OpenSSL used in binary releases per CVE-2023-0464

### ISSUE THREAD

**Issue by gpshead** [python/cpython#103142]
https://nvd.nist.gov/vuln/detail/CVE-2023-0464

We need OpenSSL >= 1.1.1u | 3.0.9 | 3.1.1.

We've got patch releases coming up soon, we should be able to just bump the versions before then.

_(note: at the time of this writing OpenSSL hasn't even released those updated versions)_

### Linked PRs
* gh-105129
* gh-105130
* gh-105131
* gh-105132
* gh-105174
* gh-105199
* gh-105200
* gh-105204
* gh-105205
* gh-105308
* gh-105370

**Comment by ambv** [comment]
We are still waiting on the upstream OpenSSL release as of today.

**Comment by JelleZijlstra** [comment]
From https://www.openssl.org/source/ it appears there's no 1.1.1u release yet. Is this still a blocker?

**Comment by gpshead** [comment]
wait until the day our releases are being built and bump it back to deferred if upstream openssl hasn't done releases.

**Comment by h-vetinari** [comment]
FYI: Their next release is [planned](https://github.com/openssl/openssl/issues/20819#issuecomment-1520051184) to be in May.

**Comment by gpshead** [comment]
I hope they do that, we're currently planning to cut 3.11.4 in the first week of June per https://peps.python.org/pep-0664/.

**Comment by h-vetinari** [comment]
> We need OpenSSL >= 1.1.1u | 3.0.9 | 3.1.1.

These were all [released](https://github.com/openssl/openssl/tags) a few hours ago.

**Comment by gpshead** [comment]
Everything back through 3.11 has been updated to build binary releases using OpenSSL 1.1.1u.

Backports for 3.10 and 3.9 exist as PRs for their respective @pablogsal and @ambv release managers to merge.

For 3.8 and 3.7 branches, I recommend doing a backport from the 3.9 commit once that PR is merged (if the RMs choose to merge it; we don't build binary releases for versions that old so that's up to them - it's at least useful for other vendors who potentially do?)

**Comment by ned-deily** [comment]
This has now been backported to 3.8 and 3.7 as well. For 3.7, there is only a partial backport because some of the changes in earlier releases to fully support OpenSSL 3.x were not backported to 3.7.

**Comment by keate01** [comment]
from PIL import Image, ImageDraw

# Créer une image vierge pour la photo de profil (taille 200x200 pixels)
profile_image = Image.new("RGB", (200, 200), color=(255, 255, 255))

# Dessiner un cercle pour simuler une photo de profil
draw = ImageDraw.Draw(profile_image)
draw.ellipse((10, 10, 190, 190), fill="blue", outline="black", width=2)

# Sauvegarder l'image de profil
profile_image.save("profile_image.jpg")

# Créer une image vierge pour la couverture (taille 1080x360 pixels)
cover_image = Image.new("RGB", (1080, 360), color=(255, 255, 255))

# Dessiner un texte sur la couverture
draw = ImageDraw.Draw(cover_image)
font_size = 50
text = "Côte d'Ivoire la Belle"
text_width, text_height = draw.textsize(text)
x = (1080 - text_width) // 2
y = (360 - text_height) // 2
draw.text((x, y), text, fill="black", font=("Arial", font_size))

# Sauvegarder l'image de couverture
cover_image.save("cover_image.jpg")
